tests/unit/test_jobs.py

```diff
@@ -0,0 +1,33 @@
+from catkin_tools.jobs.utils import merge_envs
+
+
+def test_merge_envs_basic():
+    job_env = { 'PATH': '/usr/local/bin:/usr/bin', 'FOO': 'foo' }
+
+    merge_envs(job_env, [
+        { 'PATH': '/usr/local/bin:/bar/baz/bin' },
+        { 'BAR': 'bar' } ])
+
+    # Validate that the known path was not moved from the existing order, and the unfamiliar
+    # path was correctly prepended.
+    assert job_env['PATH'] == '/bar/baz/bin:/usr/local/bin:/usr/bin'
```

Comment:
The logic is that it's introducing `/usr/local/bin` and `/bar/baz/bin` as new paths. It deduplicates first, and maintains order of new paths second, so `/usr/local/bin` is tossed as part of deduplication, and `/bar/baz/bin` is prefixed onto the existing paths.

What would be the logic that would lead to '/bar/baz/bin:/usr/local/bin:/usr/bin'?